YAML/YAML.sublime-syntax

```diff
@@ -527,16 +527,23 @@ contexts:
             - match: '{{_flow_scalar_end_plain_out}}'
               pop: true
     - match: :(?=\s|$)
-      scope: punctuation.separator.mapping.key-value.yaml
+      scope: punctuation.separator.key-value.mapping.yaml
 
   comment:
     # http://www.yaml.org/spec/1.2/spec.html#comment//
     - match: | # l-comment
         (?x)
-        (?: ^ [ \t]* | [ \t]+ )
-        (?:(\#) \p{Print}* )?
-        (\n|\z)
-      scope: comment.line.number-sign.yaml
+        (?: ( ^ [ \t]* ) | [ \t]+ )
+        (?=#\p{Print}*$)
       captures:
-        1: punctuation.definition.comment.line.number-sign.yaml
+        1: punctuation.whitespace.comment.leading.yaml
```

Comment:
In Textmate we use this scope to be able to have commands, with scope as the input, match an entire block of comment lines. This allows for things like reflowing the comment text, etc. If this isn't used in Sublime the rule can be simplified.